compiler/frontend/beam/src/main/java/edu/snu/nemo/compiler/frontend/beam/NemoPipelineVisitor.java

```diff
@@ -241,6 +241,19 @@ private static BeamCoder getCoderForView(final ViewFn viewFn, final Coder beamIn
    */
   private static DataCommunicationPatternProperty.Value getEdgeCommunicationPattern(final IRVertex src,
                                                                                     final IRVertex dst) {
+    try {
+      final Class<?> constructUnionTableFn = Class.forName(
+          "org.apache.beam.sdk.transforms.join.CoGroupByKey$ConstructUnionTableFn");
+      if (src instanceof OperatorVertex && ((OperatorVertex) src).getTransform() instanceof DoTransform
+          && ((DoTransform) ((OperatorVertex) src).getTransform()).getDoFn().getClass().equals(constructUnionTableFn)) {
+        return DataCommunicationPatternProperty.Value.Shuffle;
+      }
+    } catch (final ClassNotFoundException e) {
+      throw new RuntimeException(e);
+    }
+    if (src instanceof OperatorVertex && ((OperatorVertex) src).getTransform() instanceof FlattenTransform) {
```

Comment:
Isn't this case already handled in the code below?
If you want to make it explicit, why not add a comment or include it in the if-else statement below?

Reply:
![screenshot from 2018-05-28 14-13-24](https://user-images.githubusercontent.com/8655816/40598194-533a7b6a-6281-11e8-9b9a-297b5abe7aeb.png)

Unfortunately, no. Without that `if` block, the method will return `Shuffle` pattern.